nipype/interfaces/dipy/simulate.py

```diff
@@ -0,0 +1,274 @@
+
+        vfsim = nb.concat_images([nb.load(f) for f in self.inputs.in_vfms])
+        vfs = np.squeeze(vfsim.get_data())  # volume fractions
+
+        total_ff = np.sum(ffs, axis=3)
+        total_vf = np.sum(vfs, axis=3)
+
+        msk = np.zeros(shape, dtype=np.uint8)
+        msk[(total_vf > 0.0) & (total_ff > 0.0)] = 1
+
+        if isdefined(self.inputs.in_mask):
+            msk = nb.load(self.inputs.in_mask).get_data()
+            msk[msk > 0.0] = 1.0
+            msk[msk < 1.0] = 0.0
+
+        mhdr = hdr.copy()
+        mhdr.set_data_dtype(np.uint8)
+        mhdr.set_xyzt_units('mm', 'sec')
+        nb.Nifti1Image(msk, aff, mhdr).to_filename(
+            op.abspath(self.inputs.out_mask))
+
+        args = np.hstack((vfs[msk > 0], ffs[msk > 0]))
+
+        for f in self.inputs.in_dirs:
+            fd = nb.load(f).get_data()
+            args = np.hstack((args, fd[msk > 0]))
+
+        b0 = np.array([b0_im.get_data()[msk > 0]]).T
+        args = np.hstack((args, b0))
+
+        if isdefined(self.inputs.bval) and isdefined(self.inputs.bvec):
+            # Load the gradient strengths and directions
+            bvals = np.loadtxt(self.inputs.bval)
+            bvecs = np.loadtxt(self.inputs.bvec).T
+
+            # Place in Dipy's preferred format
+            gtab = gradient_table(bvals, bvecs)
+        else:
+            gtab = _generate_gradients(self.inputs.num_dirs,
+                                       self.inputs.bvalues)
+
+        np.savetxt(op.abspath(self.inputs.out_bvec), gtab.bvecs.T)
+        np.savetxt(op.abspath(self.inputs.out_bval), gtab.bvals.T)
+
+        snr = self.inputs.snr
+        args = [tuple(np.hstack((r, gtab, snr))) for r in args]
+
+        n_proc = self.inputs.n_proc
+        if n_proc == 0:
+            n_proc = cpu_count()
+
+        try:
+            pool = Pool(processes=n_proc, maxtasksperchild=50)
+        except TypeError:
+            pool = Pool(processes=n_proc)
+
+        iflogger.info(('Starting simulation of %d voxels, %d diffusion'
+                       ' directions.') % (len(args), len(gtab.bvals)))
+        result = pool.map(_compute_voxel, args)
+        ndirs = np.shape(result)[1]
+
+        simulated = np.zeros((shape[0], shape[1], shape[2], ndirs))
+        simulated[msk > 0] = result
+
+        simhdr = hdr.copy()
+        simhdr.set_data_dtype(np.float32)
+        simhdr.set_xyzt_units('mm', 'sec')
+        nb.Nifti1Image(simulated.astype(np.float32), aff,
+                       simhdr).to_filename(op.abspath(self.inputs.out_file))
+
+        return runtime
+
+    def _list_outputs(self):
+        outputs = self._outputs().get()
+        outputs['out_file'] = op.abspath(self.inputs.out_file)
+        outputs['out_mask'] = op.abspath(self.inputs.out_mask)
+        outputs['out_bvec'] = op.abspath(self.inputs.out_bvec)
+        outputs['out_bval'] = op.abspath(self.inputs.out_bval)
+
+        return outputs
+
+
+def _compute_voxel(args):
```

Comment:
I don't really know how this usually works in nipype, but it would be good to have clear documentation for what goes into `args`. There's lots of them!